flink/src/main/scala/ru/itclover/tsp/services/ArrowService.scala

```diff
@@ -0,0 +1,137 @@
+package ru.itclover.tsp.services
+
+import java.io.{ByteArrayInputStream, File, FileInputStream}
+
+import org.apache.arrow.memory.{BaseAllocator, RootAllocator}
+import org.apache.arrow.vector.ipc.{ArrowFileReader, ArrowReader, ArrowStreamReader, SeekableReadChannel}
+import org.apache.arrow.vector.{BaseValueVector, BitVector, Float4Vector, Float8Vector, IntVector, VarCharVector, BigIntVector, SmallIntVector}
+import org.apache.arrow.vector.types.Types
+import org.apache.arrow.vector.types.pojo.Schema
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+
+object ArrowService {
+
+  /**
+  * Method for working with types from Apache Arrow schema.
+  */
+  def typesMap: Map[Types.MinorType, Class[_ >: Float with String with Boolean with Int with Double with Long with Short]] = Map(
+    Types.MinorType.BIGINT -> classOf[Long],
+    Types.MinorType.SMALLINT -> classOf[Short],
+    Types.MinorType.BIT -> classOf[Boolean],
+    Types.MinorType.INT -> classOf[Int],
+    Types.MinorType.VARCHAR -> classOf[String],
+    Types.MinorType.FLOAT4 -> classOf[Float],
+    Types.MinorType.FLOAT8 -> classOf[Double]
+  )
+
+  /**
+  * Method for retrieving schema and reader from input bytes
+    * @param inputData byte array with input data
+    * @return tuple with schema and reader
+    */
+  def retrieveSchemaAndReader(inputData: Array[Byte]): (Schema, ArrowReader, BaseAllocator) = {
+
+    val allocator = new RootAllocator(Integer.MAX_VALUE)
+    val bytesStream = new ByteArrayInputStream(inputData)
+
+    val reader = new ArrowStreamReader(bytesStream, allocator)
+
+    (reader.getVectorSchemaRoot.getSchema, reader, allocator)
+
+  }
+
+  /**
+  * Method for retrieving schema and reader from input file
+    * @param inputData file with input data
+    * @return tuple with schema and reader
+    */
+  def retrieveSchemaAndReader(inputData: File): (Schema, ArrowReader, BaseAllocator) = {
```

Comment:
this will not be a file, but some input stream

Reply:
parquet doesn't suport read from bytes

Reply:
maybe we should send bytes to FileService, and expect file from him